### PR TITLE
(Upgrades for /depends + gitian-*.yml setup)(Fix for Coin Control "sorting" bug) (Fix for windows builds hashes) (Statically linked libcurl)

### PR DESCRIPTION
**Pull request by mxaddict**
Fix for Coin Control "sorting" bug

This is a fix for issues #427 #471 #473

### CODE REVIEW

**Comment by mxaddict** [comment]
NOTE: The fix involves upgrading to `QT 5.9.8`, which breaks compatibility for `OSX` older than `10.10`

**Comment by aguycalled** [comment]
confirmed issue fixed. built on debian and osx using the new dependencies. pending gitian build

**Comment by aguycalled** [comment]
gitian fails when doing `make -j2 -C src check-symbols`

**Comment by mxaddict** [comment]
I have a testnet wallet that has been running with this build for 1.5 days now, and I have tested sending recieving and staking (including cold staking) on testnet

I think this PR is ready for review.

**Comment by mxaddict** [comment]
@aguycalled  @craigmacgregor  I've managed to get all the related issues in order on this branch's PR

But i've noticed that GLIBC requirements for this PR require Ubuntu >= 18.04

On other distro's not sure (Anything with GLIBC >= 2.27 ABI)

I will be adding some of the functions that are causing the incompatibility to the `glibc_compat.cpp` file to resolve this.

What do you  guys think?

**Comment by mxaddict** [comment]
@aguycalled as per discussed on discord, I will leave this branch in it';s current state and will do gitian builds and post  the hashes here for comparison.

**Comment by mxaddict** [comment]
gitian hashes:

```
1d26fad1787b338e9140de5304997982261fa2055bff4a86e59444e18f6ac83e  navcoin-4.6.0-aarch64-linux-gnu-debug.tar.gz
67219cc64ab034d25b6be901818c88d3ae222da521e5b57ad984b0e459678ce9  navcoin-4.6.0-aarch64-linux-gnu.tar.gz
96b69baed13837b367a74a648fbb034ee231d7b87b60ce4b119c1f7eb3854d14  navcoin-4.6.0-arm-linux-gnueabihf-debug.tar.gz
e475e291c287e1f866f3fe03d0c057bcc8be17fe411def526c8b85d3722a497f  navcoin-4.6.0-arm-linux-gnueabihf.tar.gz
cddca3649b5f1cfaaf996f16e9887e73cd9d0caba9fe1a8679ea40e88e15f0da  navcoin-4.6.0-i686-pc-linux-gnu-debug.tar.gz
26f93b137ca4f9f668614eaabeda4b422e920ab3d2701f2c0594ef914b186e6b  navcoin-4.6.0-i686-pc-linux-gnu.tar.gz
2f9772520827241b3abbe8e77bad41e4e0ec1e687a52cacc973f570ff6385a8b  navcoin-4.6.0-x86_64-linux-gnu-debug.tar.gz
8b68afdd68457ff238f25eb0edd61569de1b7a5bf52285c1cce9ebb2c3cc3e4b  navcoin-4.6.0-x86_64-linux-gnu.tar.gz                                                                               
94c7301994f77a29f16fb62de2e3d4a34e3e070466b08c8b06a5039152f1e47d  src/navcoin-4.6.0.tar.gz
199e701aa674c03b51e75b2498e19cc734f7c9e474800b1dc36ef968d650b8dd  navcoin-linux-4.0.0-res.yml
```

**Comment by mxaddict** [comment]
Updated hashes after fixing conflict from master branch:

```
a2f931814d31c392d978f8b145d0e7c10a04fff826a8ee0528bb252a64c7fcbf  navcoin-4.6.0-aarch64-linux-gnu.tar.gz
529345176af82959a39aa1ccd93589edb1843025c17d00e961c9c1f01a6ba26e  navcoin-4.6.0-arm-linux-gnueabihf.tar.gz
a9371072b2a1931e191a8d26505b4e95e20e2b0d3984ed633b70e891346b435b  navcoin-4.6.0-i686-pc-linux-gnu.tar.gz
a55c619308766030a846a37c116d96886d674da84cc790f4a5ac9b8110499bac  navcoin-4.6.0-osx-unsigned.dmg
ca1a50867b15e0e71b483f3402dbfe447e0f3b91c8f79790a58b3280c75e2021  navcoin-4.6.0-osx-unsigned.tar.gz
033c8e83159f6eaf638b59be3e8b1b4fc792fbf0b5f0cb8bab00a2c5a634f1ac  navcoin-4.6.0-osx64.tar.gz
4bf7b84019a6a5b7bb0e08c1df76ce37101850ad62dac18ab8505785fdd93900  navcoin-4.6.0-win32-setup-unsigned.exe
2bfeaca0a9a1a9beca2fa5dff09f551b28b9d80caae5fb2fb4edd0e1bfa3ce30  navcoin-4.6.0-win32.zip
e112692aa62e48180d738ccd80a2c00b9d87afd3d445c1d2f08f0e14c35b7f5a  navcoin-4.6.0-win64-setup-unsigned.exe
a1bc0ab3cfba76bc2e3f9016b8d02fd46a9073b6a0a8d788cde2f8def6bbbc3b  navcoin-4.6.0-win64.zip
178ec34849ed4349ee998a7d0a0d2b321d2a215c0b17a0a67b164bb7dfe34f7b  navcoin-4.6.0-x86_64-linux-gnu.tar.gz
5e44a45c8791355ae3b79e0b1a40d0655908f051d52a74a98ed5527f986cc27a  navcoin-4.6.0.tar.gz
```

**Comment by aguycalled** [comment]
Hashes of my builds:

`a2f931814d31c392d978f8b145d0e7c10a04fff826a8ee0528bb252a64c7fcbf  navcoin-4.6.0-aarch64-linux-gnu.tar.gz
529345176af82959a39aa1ccd93589edb1843025c17d00e961c9c1f01a6ba26e  navcoin-4.6.0-arm-linux-gnueabihf.tar.gz
a9371072b2a1931e191a8d26505b4e95e20e2b0d3984ed633b70e891346b435b  navcoin-4.6.0-i686-pc-linux-gnu.tar.gz
d8bf8c19fd3bd15f7040c6ac018d39ea8cf05567d5b4f3ef4bbe8bfd7fdc5108  navcoin-4.6.0-osx-unsigned.dmg
d89ad4c4016834ed7ccd43cfb3a51713d5a8eab8c7f563bda3ab7593d109dd53  navcoin-4.6.0-osx-unsigned.tar.gz
6404cc63c8d9b4c9f3b67c04f3ad68a590518fb28b1db50933a78c298c322d4d  navcoin-4.6.0-osx64.tar.gz
4bf7b84019a6a5b7bb0e08c1df76ce37101850ad62dac18ab8505785fdd93900  navcoin-4.6.0-win32-setup-unsigned.exe
2bfeaca0a9a1a9beca2fa5dff09f551b28b9d80caae5fb2fb4edd0e1bfa3ce30  navcoin-4.6.0-win32.zip
e112692aa62e48180d738ccd80a2c00b9d87afd3d445c1d2f08f0e14c35b7f5a  navcoin-4.6.0-win64-setup-unsigned.exe
a1bc0ab3cfba76bc2e3f9016b8d02fd46a9073b6a0a8d788cde2f8def6bbbc3b  navcoin-4.6.0-win64.zip
178ec34849ed4349ee998a7d0a0d2b321d2a215c0b17a0a67b164bb7dfe34f7b  navcoin-4.6.0-x86_64-linux-gnu.tar.gz
5e44a45c8791355ae3b79e0b1a40d0655908f051d52a74a98ed5527f986cc27a  navcoin-4.6.0.tar.gz`

Everything matches but osx

**Comment by mxaddict** [comment]
> Hashes of my builds:
> 
> `a2f931814d31c392d978f8b145d0e7c10a04fff826a8ee0528bb252a64c7fcbf navcoin-4.6.0-aarch64-linux-gnu.tar.gz 529345176af82959a39aa1ccd93589edb1843025c17d00e961c9c1f01a6ba26e navcoin-4.6.0-arm-linux-gnueabihf.tar.gz a9371072b2a1931e191a8d26505b4e95e20e2b0d3984ed633b70e891346b435b navcoin-4.6.0-i686-pc-linux-gnu.tar.gz d8bf8c19fd3bd15f7040c6ac018d39ea8cf05567d5b4f3ef4bbe8bfd7fdc5108 navcoin-4.6.0-osx-unsigned.dmg d89ad4c4016834ed7ccd43cfb3a51713d5a8eab8c7f563bda3ab7593d109dd53 navcoin-4.6.0-osx-unsigned.tar.gz 6404cc63c8d9b4c9f3b67c04f3ad68a590518fb28b1db50933a78c298c322d4d navcoin-4.6.0-osx64.tar.gz 4bf7b84019a6a5b7bb0e08c1df76ce37101850ad62dac18ab8505785fdd93900 navcoin-4.6.0-win32-setup-unsigned.exe 2bfeaca0a9a1a9beca2fa5dff09f551b28b9d80caae5fb2fb4edd0e1bfa3ce30 navcoin-4.6.0-win32.zip e112692aa62e48180d738ccd80a2c00b9d87afd3d445c1d2f08f0e14c35b7f5a navcoin-4.6.0-win64-setup-unsigned.exe a1bc0ab3cfba76bc2e3f9016b8d02fd46a9073b6a0a8d788cde2f8def6bbbc3b navcoin-4.6.0-win64.zip 178ec34849ed4349ee998a7d0a0d2b321d2a215c0b17a0a67b164bb7dfe34f7b navcoin-4.6.0-x86_64-linux-gnu.tar.gz 5e44a45c8791355ae3b79e0b1a40d0655908f051d52a74a98ed5527f986cc27a navcoin-4.6.0.tar.gz`
> 
> Everything matches but osx

Hmmm, what SDK did you use for the OSX build?

**Comment by mxaddict** [comment]
This is the SDK that I used:

```
curl --location --fail https://bitcoincore.org/depends-sources/sdks/MacOSX10.11.sdk.tar.gz -o inputs/MacOSX10.11.sdk.tar.gz
```

**Comment by aguycalled** [comment]
```$ sha256sum MacOSX10.11.sdk.tar.gz 
bc1cf275e3cc675e820aecb72f98f587257d59fcf27f8dfafd4a176430594be9  MacOSX10.11.sdk.tar.gz```

**Comment by mxaddict** [comment]
```
bec9d089ebf2e2dd59b1a811a38ec78ebd5da18cbbcd6ab39d1e59f64ac5033f  ../gitian-builder/inputs/MacOSX10.11.sdk.tar.gz
```

**Comment by aguycalled** [comment]
will get it from bitcoincore and relaunch mac build

**Comment by mxaddict** [comment]
> will get it from bitcoincore and relaunch mac build

ACK

**Comment by aguycalled** [comment]
```a55c619308766030a846a37c116d96886d674da84cc790f4a5ac9b8110499bac  navcoin-4.6.0-osx-unsigned.dmg
ca1a50867b15e0e71b483f3402dbfe447e0f3b91c8f79790a58b3280c75e2021  navcoin-4.6.0-osx-unsigned.tar.gz
033c8e83159f6eaf638b59be3e8b1b4fc792fbf0b5f0cb8bab00a2c5a634f1ac  navcoin-4.6.0-osx64.tar.gz```

bingo

**Comment by aguycalled** [comment]
tested correctly executed in enviroment without libcurl

**Comment by aguycalled** [comment]
tested locally in debian

File: configure.ac [hunk 1]
```diff
@@ -504,12 +504,6 @@ if test x$TARGET_OS != xwindows; then
   # All windows code is PIC, forcing it on just adds useless compile warnings
   AX_CHECK_COMPILE_FLAG([-fPIC],[CFLAGS="-fPIC"])
 
-  # check to see if we are compiling on a x86_64-pc-linux-gnu the force fPIC for QT Build compliance
-  if test x$host == xx86_64-pc-linux-gnu ; then
-    # All windows code is PIC, forcing it on just adds useless compile warnings
-    AX_CHECK_COMPILE_FLAG([-fPIC],[PIC_FLAGS="-fPIC"])
-   fi
-
```
[the remark this answers is not in the document]
Comment (mxaddict) [reply configure.ac]: May have been an oversight on my part when merging upstream changes.

I'll re-add it.

[the remark this answers is not in the document]
Comment (aguycalled) [reply configure.ac]: i remember seeing this alert when building the gui in some systems, but maybe it's not necessary anymore with the new qt version

[the remark this answers is not in the document]
Comment (mxaddict) [reply configure.ac]: NVM, it was not removed, it was MOVED :D

[the remark this answers is not in the document]
Comment (mxaddict) [reply configure.ac]: This was removed:
```
AX_CHECK_COMPILE_FLAG([-fPIC],[CFLAGS="-fPIC"])
```

And 
```
AX_CHECK_COMPILE_FLAG([-fPIC],[PIC_FLAGS="-fPIC"])
```

Was put in it's place.

**Comment by mxaddict** [comment]
> tested correctly executed in enviroment without libcurl

Were you able to also test the original issue for this PR?

Sorting of the inputs on coin control dialog :+1:

**Comment by aguycalled** [comment]
yes, the issue is fixed for me

**Comment by craigmacgregor** [comment]
compiling on linux now. do you want me to run a full gititan build and compare hashes?

**Comment by mxaddict** [comment]
> compiling on linux now. do you want me to run a full gititan build and compare hashes?

Yes please, that is one of the fixes, all build hashes should match :+1:

**Comment by craigmacgregor** [comment]
my linux build finished and the hashes match

**Comment by craigmacgregor** [comment]
Linux hashes
```
600fffa7e6072b4062c404ba54aa68ba002717ee5dce26bbbca95e6428281439  build/out/navcoin-4.6.0-aarch64-linux-gnu-debug.tar.gz
a2f931814d31c392d978f8b145d0e7c10a04fff826a8ee0528bb252a64c7fcbf  build/out/navcoin-4.6.0-aarch64-linux-gnu.tar.gz
25a7dc2fea99fc8b605a2cc694d73bda367589c6436c2542956ac9e423460b5c  build/out/navcoin-4.6.0-arm-linux-gnueabihf-debug.tar.gz
529345176af82959a39aa1ccd93589edb1843025c17d00e961c9c1f01a6ba26e  build/out/navcoin-4.6.0-arm-linux-gnueabihf.tar.gz
778e92dd93cf725178a50c42be2be290ea83a2f14f44b69d791528c9e2c1a4c0  build/out/navcoin-4.6.0-i686-pc-linux-gnu-debug.tar.gz
a9371072b2a1931e191a8d26505b4e95e20e2b0d3984ed633b70e891346b435b  build/out/navcoin-4.6.0-i686-pc-linux-gnu.tar.gz
ee9581e9fbae17a2c876816a3494df67e530ed59b8c948a51df08de281ee0ede  build/out/navcoin-4.6.0-x86_64-linux-gnu-debug.tar.gz
178ec34849ed4349ee998a7d0a0d2b321d2a215c0b17a0a67b164bb7dfe34f7b  build/out/navcoin-4.6.0-x86_64-linux-gnu.tar.gz
```

OSX hashes
```
a55c619308766030a846a37c116d96886d674da84cc790f4a5ac9b8110499bac  build/out/navcoin-4.6.0-osx-unsigned.dmg
ca1a50867b15e0e71b483f3402dbfe447e0f3b91c8f79790a58b3280c75e2021  build/out/navcoin-4.6.0-osx-unsigned.tar.gz
033c8e83159f6eaf638b59be3e8b1b4fc792fbf0b5f0cb8bab00a2c5a634f1ac  build/out/navcoin-4.6.0-osx64.tar.gz
```

Windows hashes
```
b1dd88cdd8bf09af2539d0b345e647129d8fd55c92b6d824b6ecc53efd531028  build/out/navcoin-4.6.0-win.tar.gz
11f3e4170f104c2ffc78e08603a03df7a9be22fe72431932e4063d42412d230b  build/out/navcoin-4.6.0-win32-debug.zip
4bf7b84019a6a5b7bb0e08c1df76ce37101850ad62dac18ab8505785fdd93900  build/out/navcoin-4.6.0-win32-setup-unsigned.exe
2bfeaca0a9a1a9beca2fa5dff09f551b28b9d80caae5fb2fb4edd0e1bfa3ce30  build/out/navcoin-4.6.0-win32.zip
df559db6d69bb4b01a69bd8dd981bbac947dee781087e0f567772feb8f54227f  build/out/navcoin-4.6.0-win64-debug.zip
e112692aa62e48180d738ccd80a2c00b9d87afd3d445c1d2f08f0e14c35b7f5a  build/out/navcoin-4.6.0-win64-setup-unsigned.exe
a1bc0ab3cfba76bc2e3f9016b8d02fd46a9073b6a0a8d788cde2f8def6bbbc3b  build/out/navcoin-4.6.0-win64.zip
```

All matching

**Pull request merged**